client.go

```diff
@@ -944,7 +944,7 @@ func (f *File) Read(b []byte) (int, error) {
 // readChunkAt attempts to read the whole entire length of the buffer from the file starting at the offset.
 // It will continue progressively reading into the buffer until it fills the whole buffer, or an error occurs.
 func (f *File) readChunkAt(ch chan result, b []byte, off int64) (n int, err error) {
-	for err == nil && n < len(b) {
+	for n < len(b) {
```

Comment:
This `err == nil` was always true, because line 949 shadowed the variable. Since every error path already `return`s in the loop anyways, the check was entirely extraneous.